src/network/tcp.py

```diff
@@ -113,50 +117,50 @@ def set_connection_fully_established(self):
         addrs = {}
         for stream in self.streams:
             with knownnodes.knownNodesLock:
-                if len(knownnodes.knownNodes[stream]) > 0:
+                if knownnodes.knownNodes[stream]:
                     filtered = {k: v for k, v in knownnodes.knownNodes[stream].items()
-                        if v["lastseen"] > (int(time.time()) - shared.maximumAgeOfNodesThatIAdvertiseToOthers)}
+                                if v["lastseen"] > (int(time.time()) - shared.maximumAgeOfNodesThatIAdvertiseToOthers)}
                     elemCount = len(filtered)
                     if elemCount > maxAddrCount:
                         elemCount = maxAddrCount
                     # only if more recent than 3 hours
                     addrs[stream] = helper_random.randomsample(filtered.items(), elemCount)
                 # sent 250 only if the remote isn't interested in it
-                if len(knownnodes.knownNodes[stream * 2]) > 0 and stream not in self.streams:
-                    filtered = {k: v for k, v in knownnodes.knownNodes[stream*2].items()
-                        if v["lastseen"] > (int(time.time()) - shared.maximumAgeOfNodesThatIAdvertiseToOthers)}
+                if knownnodes.knownNodes[stream * 2] and stream not in self.streams:
+                    filtered = {k: v for k, v in knownnodes.knownNodes[stream * 2].items()
+                                if v["lastseen"] > (int(time.time()) - shared.maximumAgeOfNodesThatIAdvertiseToOthers)}
                     elemCount = len(filtered)
                     if elemCount > maxAddrCount / 2:
                         elemCount = int(maxAddrCount / 2)
                     addrs[stream * 2] = helper_random.randomsample(filtered.items(), elemCount)
-                if len(knownnodes.knownNodes[(stream * 2) + 1]) > 0 and stream not in self.streams:
-                    filtered = {k: v for k, v in knownnodes.knownNodes[stream*2+1].items()
-                        if v["lastseen"] > (int(time.time()) - shared.maximumAgeOfNodesThatIAdvertiseToOthers)}
+                if knownnodes.knownNodes[(stream * 2) + 1] and stream not in self.streams:
+                    filtered = {k: v for k, v in knownnodes.knownNodes[stream * 2 + 1].items()
+                                if v["lastseen"] > (int(time.time()) - shared.maximumAgeOfNodesThatIAdvertiseToOthers)}
                     elemCount = len(filtered)
                     if elemCount > maxAddrCount / 2:
                         elemCount = int(maxAddrCount / 2)
                     addrs[stream * 2 + 1] = helper_random.randomsample(filtered.items(), elemCount)
-        for substream in addrs.keys():
+        for substream in addrs:
             for peer, params in addrs[substream]:
                 templist.append((substream, peer, params["lastseen"]))
-        if len(templist) > 0:
+        if templist:
             self.append_write_buf(BMProto.assembleAddr(templist))
 
     def sendBigInv(self):
+        """TBC"""
```

Comment:
Send hashes of all inventory objects, chunked as the protocol has a per-command limit.